src/conversion_driver_esm1p5.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+ESM1.5 conversion driver
+
+Wrapper script for automated fields file to NetCDF conversion
+during ESM1.5 simulations. Runs conversion module (currently 
+um2netcdf4) on each atmospheric output in a specified directory.
+
+Adapted from Martin Dix's conversion driver for CM2: 
+https://github.com/ACCESS-NRI/access-cm2-drivers/blob/main/src/run_um2netcdf.py
+"""
+
+
+
+import os
+import collections
+import um2netcdf4
+import re
+import f90nml
+import warnings
+import argparse
+import errno
+from pathlib import Path
+from collections.abc import Sequence
+
+
+def convert_esm1p5_dir(FF_dir, NC_dir, FF_name_pattern):
+    """
+    Convert ESM1.5 fields file outputs in specified directory to netCDF.
+    
+    Parameters
+    ----------
+    FF_dir : Path to source directory containing UM fields files for conversion.
+    NC_dir : Path to target directory for saving NetCDF files.
+    FF_naming_pattern : Regex pattern. Files with matching names will be converted.
+
+    Returns
+    -------
+    None
+    """
+
+    if isinstance(FF_dir, str):
+        FF_dir = Path(FF_dir)
+
+    # Find output fields files in the specified directory
+    output_fields_files = [
+        filename
+        for filename in os.listdir(FF_dir)
+        if re.match(FF_name_pattern, filename)
+    ]
+
+    for FF_name in output_fields_files:
+        FF_file_path = FF_dir / FF_name
+        NC_name = FF_name + ".nc"
+        NC_file_path = NC_dir / NC_name
+
+        # Named tuple to hold the argument list
+        Args = collections.namedtuple(
```

Comment:
Do you mind adding a `TODO` here, referencing issue #17?

For the switchover, the replacement `um2nc` should implement `kwargs` & automatically default this arg list.